sky/resources.py

```diff
@@ -436,6 +431,18 @@ def _try_validate_image_id(self) -> None:
                         'image_id is only supported for AWS in a specific '
                         'region, please explicitly specify the region.')
 
+        # Validate the image exists and the size is smaller than the disk size.
+        for region, image_id in self._image_id.items():
+            # Check the image exists and get the image size.
+            # It will raise ValueError if the image does not exist.
+            image_size = self.cloud.get_image_size(image_id, region)
+            if image_size > self.disk_size:
+                with ux_utils.print_exception_no_traceback():
+                    raise ValueError(
+                        f'Image {image_id!r} is {image_size}GB, which is '
+                        f'larger than the specified disk_size: {self.disk_size}'
+                        ' GB. Please specify a larger disk_size.')
```

Comment:
nit: `Please specify a larger disk_size to use this image`

Reply:
Done. Thanks!